### ISSUE TITLE
Put is not supported

### ISSUE THREAD

**Issue by donpatrice** [mdavidsaver/p4p#31]
Using Python 3.7 and p4p 3.1.1 I have the following problem.

```
import time
from p4p.nt import NTScalar
from p4p.server import Server, StaticProvider
from p4p.server.thread import SharedPV

pv = SharedPV(nt=NTScalar('d'), # scalar double
              initial=0.0)      # setting initial value also open()'s
provider = StaticProvider('arbitrary')
provider.add('demo:pv:name', pv) # PV name only appears here

Server.forever(providers=[provider]) # help which runs until KeyboardInterrupt
```

Then

```
$ python -m p4p.client.cli put demo:pv:name=5.0
demo:pv:name=5.0 Error: Put not supported
```

The same here
```
In [2]: from p4p.client.thread import Context

In [3]: ctxt = Context('pva')

In [4]: ctxt.put('demo:pv:name', 5)                                                                                                         
---------------------------------------------------------------------------
RemoteError                               Traceback (most recent call last)
<ipython-input-4-9b34f072a5fb> in <module>
----> 1 ctxt.put('demo:pv:name', 5)

~/.local/share/virtualenvs/cryvisil-daq-939BQ-ad/lib/python3.7/site-packages/p4p/client/thread.py in put(self, name, values, request, timeout, throw, process, wait, get)
    370                     break
    371                 if throw and isinstance(value, Exception):
--> 372                     raise value
    373                 result[i] = value
    374 

RemoteError: Put not supported
```

**Comment by mdavidsaver** [comment]
This message is trying to communicate (to a developer) that a Put handler
function needs to be provided for the SharedPV.  This can be accomplished
either by passing a Handler object the constructor, or by assigning a free
function to `pv.put =`.

**Comment by donpatrice** [comment]
Does that mean that the documentation is outdated? I cannot find a working example how to actually use that in the docs. Thanks!

**Comment by mdavidsaver** [comment]
Mostly it means that I typod (it's early morning for me).  The free function syntax is actually `@pv.put` which is shown in the example at the top of [the server page](https://mdavidsaver.github.io/p4p/server.html).  The server handler interface is [also mentioned](https://mdavidsaver.github.io/p4p/server.html#sharedpv-handler-interface)

**Comment by donpatrice** [comment]
Even with
```
import time
from p4p.nt import NTScalar
from p4p.server import Server, StaticProvider
from p4p.server.thread import SharedPV

pv = SharedPV(nt=NTScalar('d'), # scalar double
              initial=0.0)      # setting initial value also open()'s
@pv.put
def handle(pv, op):
    pv.post(op) # just store and update subscribers
    op.done()

provider = StaticProvider('arbitrary')
provider.add('demo:pv:name', pv) # PV name only appears here

Server.forever(providers=[provider]) # help which runs until KeyboardInterrupt
```
I get
```
$ python -m p4p.client.cli put demo:pv:name=5.0
demo:pv:name=5.0 Error: Can't assign scalar field value with ServOpWrap
```

**Comment by mdavidsaver** [comment]
Ah, now this is a documentation bug (not outdated, just wrong).  It should be `pv.post(op.value())`.

```
@pv.put
def handle(pv, op):
    pv.post(op.value()) # just store and update subscribers
    op.done()
```

A Handler object demonstrates this here:

https://github.com/mdavidsaver/p4p/blob/b9a4caf43516459c1c2a482e22873ff744267d5b/example/mailbox_server.py#L48-L51

**Comment by donpatrice** [comment]
Yes, that's working. Thanks! But this also means, that the handle is necessary and there is no sensible default, isn't it?

**Comment by donpatrice** [comment]
But this handler does not work with an NTTable, does it?

**Comment by mdavidsaver** [comment]
I've corrected the example in the documentation (1809619f2aaf8d1d62706d38ceccd028b8615227).

> But this handler does not work with an NTTable, does it?

Looking at it again, the NTTable helper is likely the problem.  It needs some work.

My original thinking with NTTable was to have something which could be adapted from the DB-API a la. [sqlite3](https://docs.python.org/2.7/library/sqlite3.html).  However, at the time I was only thinking about the client side.  That `NTTable.unwrap()` returns an iterator is less than helpful for a server.

In: `pv.post(op.value())` when `SharedPV(nt=NTTable(...))` the `op.value()` is returning the result of `NTTable.unwrap()`, and `pv.post()` is passing it's argument through `NTTable.wrap()`.

FYI, if one of the NT helper classes isn't helpful enough, or simply broken, you can always use the `buildType()` method to get a `Type` and handle mapping to/from `Value` objects yourself.
